kong/db/errors.lua

```diff
@@ -24,14 +24,16 @@ end
 
 
 local ERRORS            = {
-  INVALID_PRIMARY_KEY   = 1,
-  SCHEMA_VIOLATION      = 2,
-  PRIMARY_KEY_VIOLATION = 3, -- primary key already exists (HTTP 400)
-  FOREIGN_KEY_VIOLATION = 4, -- foreign entity does not exist (HTTP 400)
-  UNIQUE_VIOLATION      = 5, -- unique key already exists (HTTP 409)
-  NOT_FOUND             = 6, -- WHERE clause leads nowhere (HTTP 404)
-  INVALID_OFFSET        = 7, -- page(size, offset) is invalid
-  DATABASE_ERROR        = 8, -- connection refused or DB error (HTTP 500)
+  INVALID_PRIMARY_KEY   =  1,
+  SCHEMA_VIOLATION      =  2,
+  PRIMARY_KEY_VIOLATION =  3, -- primary key already exists (HTTP 400)
+  FOREIGN_KEY_VIOLATION =  4, -- foreign entity does not exist (HTTP 400)
+  UNIQUE_VIOLATION      =  5, -- unique key already exists (HTTP 409)
+  NOT_FOUND             =  6, -- WHERE clause leads nowhere (HTTP 404)
+  INVALID_OFFSET        =  7, -- page(size, offset) is invalid
+  DATABASE_ERROR        =  8, -- connection refused or DB error (HTTP 500)
+  CONFLICTING_INPUT     =  9, -- user-provided data generated a conflict (HTTP 409)
```

Comment:
Let's maybe make the description more precise: "a conflict _with already inserted DB values_"?

I recall discussing this with @bungle, and since this will result from `UNIQUE` constraints set in PostgreSQL, we wanted to name this value `UNIQUE_VIOLATION` to stay consistent with other `_VIOLATION` error types we already defined.

Since the certs/sni entities are already abnormal in the sense that they validate their own values and produce their own errors, I would rather avoid spreading that abnormality to the errors as well. So in other words, the conflict errors produced by certs/sni validations are not because of DB/schema UNIQUE constraints, but I don't think that matters and I think they can throw `UNIQUE_VIOLATION` which has the same semantics.

Reply:
I have chosen to use schema violations instead. The errors always happen in the context of the `snis` magical meta-param, so it "fits" to just use schema violations for those (so they are scoped inside `snis`).